dali/pipeline/operators/crop/crop.cu

```diff
@@ -28,7 +28,11 @@ __global__ void BatchedCropKernel(const int C, const int* height, const int* wid
   const int H = height[n];
   const int in_stride = in_strides[n];
   const uint8 *input_ptr = img_ptrs[n];
-  Out *output_ptr = out + (n * C * H * W);
+
+  Out *output_ptr = out;
+  for (int i = 0; i < n; i++) {
```

Comment:
I wonder how much perf we can get if we precalculate this only once and store somewhere.
We can do it on host and transfer to the GPU, or use some shared memory and store it there once.

Reply:
resolved. This index is calculated differently now